File: java/javafind/src/test/java/javafind/FinderTest.java
package javafind;

import org.junit.jupiter.api.Test;

import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.*;

public class FinderTest {

    public FinderTest() {}

    private static FindSettings getSettings() {
        var settings = new FindSettings();
        settings.addPath(".");
        return settings;
    }

    /*************************************************************
     * isMatchingDir tests
     *************************************************************/
    @Test
    public final void testIsMatchingDir_SingleDot_True() {
        var settings = getSettings();
        var finder = new Finder(settings);
        assertTrue(finder.isMatchingDir(Paths.get(".")));
    }

    @Test
    public final void testIsMatchingDir_DoubleDot_True() {
        var settings = getSettings();
        var finder = new Finder(settings);
        assertTrue(finder.isMatchingDir(Paths.get("..")));
    }

    @Test
    public final void testIsMatchingDir_IsHidden_False() {
        var settings = getSettings();
        var finder = new Finder(settings);
        assertFalse(finder.isMatchingDir(Paths.get(".git")));
    }

    @Test
    public final void testIsMatchingDir_IsHiddenIncludeHidden_True() {
        var settings = getSettings();
        settings.setIncludeHidden(true);
        var finder = new Finder(settings);
        assertTrue(finder.isMatchingDir(Paths.get(".git")));
    }

    @Test
    public final void testIsMatchingDir_NoPatterns_True() {
        var settings = getSettings();
        var finder = new Finder(settings);
        assertTrue(finder.isMatchingDir(Paths.get("/Users")));
    }

    @Test
    public final void testIsMatchingDir_MatchesInPattern_True() {
        var settings = getSettings();
        settings.addInDirPattern("Find");
        var finder = new Finder(settings);
        assertTrue(finder.isMatchingDir(Paths.get("CsFind")));
    }

    @Test
    public final void testIsMatchingDir_MatchesOutPattern_False() {
        var settings = getSettings();
        settings.addOutDirPattern("Find");
        var finder = new Finder(settings);
        assertFalse(finder.isMatchingDir(Paths.get("CsFind")));
    }

    @Test
    public final void testIsMatchingDir_DoesNotMatchInPattern_False() {
        var settings = getSettings();
        settings.addInDirPattern("FindFiles");
        var finder = new Finder(settings);
        assertFalse(finder.isMatchingDir(Paths.get("CsFind")));
    }

    @Test
    public final void testIsMatchingDir_DoesNotMatchOutPattern_True() {
        var settings = getSettings();
        settings.addOutDirPattern("FindFiles");
        var finder = new Finder(settings);
        var dir = Paths.get("CsFind");
        assertTrue(finder.isMatchingDir(dir));
    }

    @Test
    public final void testIsMatchingDir_DoesNotMatchOutPattern2_True() {
        var settings = getSettings();
        settings.addOutDirPattern("FindFiles");
        var finder = new Finder(settings);
        var dir = Paths.get("/Users/cary/src/xfind/java/javafind/ssrc/main/java/javafind");
        assertTrue(finder.isMatchingDir(dir));
    }

    /*************************************************************
     * isMatchingFileResult tests
     *************************************************************/
    @Test
    public final void testIsMatchingFileResult_NoExtensionsNoPatterns_True() {
        var settings = getSettings();
        var finder = new Finder(settings);
        var path = Paths.get("./FileUtil.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertTrue(finder.isMatchingFileResult(fileResult));
    }

    @Test
    public final void testIsMatchingFileResult_MatchesInExtension_True() {
        var settings = getSettings();
        settings.addInExtension("cs");
        var finder = new Finder(settings);
        var path = Paths.get("./FileUtil.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertTrue(finder.isMatchingFileResult(fileResult));
    }

    @Test
    public final void testIsMatchingFileResult_DoesNotMatchInExtension_False() {
        var settings = getSettings();
        settings.addInExtension("java");
        var finder = new Finder(settings);
        var path = Paths.get("./FileUtil.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertFalse(finder.isMatchingFileResult(fileResult));
    }


    @Test
    public final void testIsMatchingFileResult_MatchesOutExtension_False() {
        var settings = getSettings();
        settings.addOutExtension("cs");
        var finder = new Finder(settings);
        var path = Paths.get("./FileUtil.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertFalse(finder.isMatchingFileResult(fileResult));
    }

    @Test
    public final void testIsMatchingFileResult_DoesNotMatchOutExtension_True() {
        var settings = getSettings();
        settings.addOutExtension("java");
        var finder = new Finder(settings);
        var path = Paths.get("./FileUtil.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertTrue(finder.isMatchingFileResult(fileResult));
    }

    @Test
    public final void testIsMatchingFileResult_MatchesInPattern_True() {
        var settings = getSettings();
        settings.addInFilePattern("Find");
        var finder = new Finder(settings);
        var path = Paths.get("./Finder.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertTrue(finder.isMatchingFileResult(fileResult));
    }

    @Test
    public final void testIsMatchingFileResult_DoesNotMatchInPattern_False() {
        var settings = getSettings();
        settings.addInFilePattern("Find");
        var finder = new Finder(settings);
        var path = Paths.get("./FileUtil.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertFalse(finder.isMatchingFileResult(fileResult));
    }

    @Test
    public final void testIsMatchingFileResult_MatchesOutPattern_False() {
        var settings = getSettings();
        settings.addOutFilePattern("Find");
        var finder = new Finder(settings);
        var path = Paths.get("./Finder.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertFalse(finder.isMatchingFileResult(fileResult));
    }

    @Test
    public final void testIsMatchingFileResult_DoesNotMatchOutPattern_True() {
        var settings = getSettings();
        settings.addOutFilePattern("Find");
        var finder = new Finder(settings);
        var path = Paths.get("./FileUtil.cs");
        var fileResult = new FileResult(path, FileType.CODE);
        assertTrue(finder.isMatchingFileResult(fileResult));
    }

    /*************************************************************
     * isMatchingArchiveFile tests
     *************************************************************/
    @Test
    public final void testIsMatchingArchiveFile_NoExtensionsNoPatterns_True() {
        var settings = getSettings();
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.isMatchingArchiveFile(path));
    }

    @Test
    public final void testIsMatchingArchiveFile_MatchesInExtension_True() {
        var settings = getSettings();
        settings.addInArchiveExtension("zip");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.isMatchingArchiveFile(path));
    }

    @Test
    public final void testIsMatchingArchiveFile_DoesNotMatchInExtension_False() {
        var settings = getSettings();
        settings.addInArchiveExtension("gz");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertFalse(finder.isMatchingArchiveFile(path));
    }


    @Test
    public final void testIsMatchingArchiveFile_MatchesOutExtension_False() {
        var settings = getSettings();
        settings.addOutArchiveExtension("zip");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertFalse(finder.isMatchingArchiveFile(path));
    }

    @Test
    public final void testIsMatchingArchiveFile_DoesNotMatchOutExtension_True() {
        var settings = getSettings();
        settings.addOutArchiveExtension("gz");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.isMatchingArchiveFile(path));
    }

    @Test
    public final void testIsMatchingArchiveFile_MatchesInPattern_True() {
        var settings = getSettings();
        settings.addInArchiveFilePattern("arch");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.isMatchingArchiveFile(path));
    }

    @Test
    public final void testIsMatchingArchiveFile_DoesNotMatchInPattern_False() {
        var settings = getSettings();
        settings.addInArchiveFilePattern("archives");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertFalse(finder.isMatchingArchiveFile(path));
    }

    @Test
    public final void testIsMatchingArchiveFile_MatchesOutPattern_False() {
        var settings = getSettings();
        settings.addOutArchiveFilePattern("arch");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertFalse(finder.isMatchingArchiveFile(path));
    }

    @Test
    public final void testIsMatchingArchiveFile_DoesNotMatchOutPattern_True() {
        var settings = getSettings();
        settings.addOutArchiveFilePattern("archives");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.isMatchingArchiveFile(path));
    }

    /*************************************************************
     * filterToFileResult tests
     *************************************************************/
    @Test
    public final void testFilterToFileResult_IsHidden_Null() {
        var settings = getSettings();
        var finder = new Finder(settings);
        var path = Paths.get(".gitignore");
        assertFalse(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_IsHiddenIncludeHidden_NotNull() {
        var settings = getSettings();
        settings.setIncludeHidden(true);
        var finder = new Finder(settings);
        var path = Paths.get(".gitignore");
        assertTrue(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_ArchiveNoFindArchives_Null() {
        var settings = getSettings();
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertFalse(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_ArchiveFindArchives_NotNull() {
        var settings = getSettings();
        settings.setIncludeArchives(true);
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_IsMatchingArchiveFile_NotNull() {
        var settings = getSettings();
        settings.setIncludeArchives(true);
        settings.addInArchiveExtension("zip");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_NotIsMatchingArchiveFile_Null() {
        var settings = getSettings();
        settings.setIncludeArchives(true);
        settings.addOutArchiveExtension("zip");
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertFalse(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_ArchiveFileArchivesOnly_NotNull() {
        var settings = getSettings();
        settings.setArchivesOnly(true);
        var finder = new Finder(settings);
        var path = Paths.get("archive.zip");
        assertTrue(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_NoExtensionsNoPatterns_NotNull() {
        var settings = getSettings();
        var finder = new Finder(settings);
        var path = Paths.get("FileUtil.cs");
        assertTrue(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_IsMatchingFile_NotNull() {
        var settings = getSettings();
        settings.addInExtension("cs");
        var finder = new Finder(settings);
        var path = Paths.get("FileUtil.cs");
        assertTrue(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_NotIsMatchingFile_Null() {
        var settings = getSettings();
        settings.addOutExtension("cs");
        var finder = new Finder(settings);
        var path = Paths.get("FileUtil.cs");
        assertFalse(finder.filterToFileResult(path).isPresent());
    }

    @Test
    public final void testFilterToFileResult_NonArchiveFileArchivesOnly_Null() {
        var settings = getSettings();
        settings.setArchivesOnly(true);
        var finder = new Finder(settings);
        var path = Paths.get("FileUtil.cs");
        assertFalse(finder.filterToFileResult(path).isPresent());
    }
}

File: javascript/jsfind/tests/finder.test.js
/*
 * finder_test.js
 *
 * Some tests of finder.js
 */

const path = require('path');

const config = require('../src/config');
const {FileType} = require("../src/filetype");
const {Finder} = require('../src/finder');
const {FindSettings} = require('../src/findsettings');
const { FileResult } = require('../src/fileresult')
// const expect = require("expect");

const getSettings = () => {
    let settings = new FindSettings();
    settings.paths.push('.');
    return settings;
};

describe('testing finder', () => {
    /*************************************************************
     * isMatchingDir tests
     *************************************************************/
    it('testIsMatchingDir_SingleDot_True', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('.')).toBeTruthy();
    });

    it('testIsMatchingDir_DoubleDot_True', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('..')).toBeTruthy();
    });

    it('testIsMatchingDir_IsHidden_False', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('.git')).toBeFalsy();
    });

    it('testIsMatchingDir_IsHiddenIncludeHidden_True', () => {
        const settings = getSettings();
        settings.includeHidden = true;
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('.git')).toBeTruthy();
    });

    it('testIsMatchingDir_NoPatterns_True', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('/Users')).toBeTruthy();
    });

    it('testIsMatchingDir_MatchesInPattern_True', () => {
        const settings = getSettings();
        settings.addInDirPatterns('Find');
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('CsFind')).toBeTruthy();
    });

    it('testIsMatchingDir_MatchesOutPattern_False', () => {
        const settings = getSettings();
        settings.addOutDirPatterns('Find');
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('CsFind')).toBeFalsy();
    });

    it('testIsMatchingDir_DoesNotMatchInPattern_False', () => {
        const settings = getSettings();
        settings.addInDirPatterns('FindFiles');
        const finder = new Finder(settings);
        expect(finder.isMatchingDir('CsFind')).toBeFalsy();
    });

    it('testIsMatchingDir_DoesNotMatchOutPattern_True', () => {
        const settings = getSettings();
        settings.addOutDirPatterns('FindFiles');
        const finder = new Finder(settings);
        const dir = 'CsFind';
        expect(finder.isMatchingDir(dir)).toBeTruthy();
    });

    /*************************************************************
     * isMatchingFileResult tests
     *************************************************************/
    it('testIsMatchingFileResult_NoExtensionsNoPatterns_True', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'FileUtil.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingFileResult_MatchesInExtension_True', () => {
        const settings = getSettings();
        settings.addInExtensions('cs');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'FileUtil.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingFileResult_DoesNotMatchInExtension_False', () => {
        const settings = getSettings();
        settings.addInExtensions('java');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'FileUtil.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingFileResult_MatchesOutExtension_False', () => {
        const settings = getSettings();
        settings.addOutExtensions('cs');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'FileUtil.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingFileResult_DoesNotMatchOutExtension_True', () => {
        const settings = getSettings();
        settings.addOutExtensions('java');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'FileUtil.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingFileResult_MatchesInPattern_True', () => {
        const settings = getSettings();
        settings.addInFilePatterns('Find');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'Finder.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingFileResult_DoesNotMatchInPattern_False', () => {
        const settings = getSettings();
        settings.addInFilePatterns('Find');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'FileUtil.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingFileResult_MatchesOutPattern_False', () => {
        const settings = getSettings();
        settings.addOutFilePatterns('Find');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'Finder.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingFileResult_DoesNotMatchOutPattern_True', () => {
        const settings = getSettings();
        settings.addOutFilePatterns('Find');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'FileUtil.cs', FileType.CODE, 0, 0);
        expect(finder.isMatchingFileResult(fr)).toBeTruthy();
    });

    /*************************************************************
     * testIsMatchingArchiveFileResult tests
     *************************************************************/
    it('testIsMatchingArchiveFileResult_NoExtensionsNoPatterns_True', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingArchiveFileResult_MatchesInExtension_True', () => {
        const settings = getSettings();
        settings.addInArchiveExtensions('zip');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingArchiveFileResult_DoesNotMatchInExtension_False', () => {
        const settings = getSettings();
        settings.addInArchiveExtensions('gz');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingArchiveFileResult_MatchesOutExtension_False', () => {
        const settings = getSettings();
        settings.addOutArchiveExtensions('zip');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingArchiveFileResult_DoesNotMatchOutExtension_True', () => {
        const settings = getSettings();
        settings.addOutArchiveExtensions('gz');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingArchiveFileResult_MatchesInPattern_True', () => {
        const settings = getSettings();
        settings.addInArchiveFilePatterns('arch');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeTruthy();
    });

    it('testIsMatchingArchiveFileResult_DoesNotMatchInPattern_False', () => {
        const settings = getSettings();
        settings.addInArchiveFilePatterns('archives');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingArchiveFileResult_MatchesOutPattern_False', () => {
        const settings = getSettings();
        settings.addOutArchiveFilePatterns('arch');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeFalsy();
    });

    it('testIsMatchingArchiveFileResult_DoesNotMatchOutPattern_True', () => {
        const settings = getSettings();
        settings.addOutArchiveFilePatterns('archives');
        const finder = new Finder(settings);
        const fr = new FileResult('.', 'archive.zip', FileType.ARCHIVE, 0, 0);
        expect(finder.isMatchingArchiveFileResult(fr)).toBeTruthy();
    });

    /*************************************************************
     * filterToFileResult tests
     *************************************************************/
    it('testFilterToFileResult_IsHidden_False', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        const file = '.gitignore';
        finder.filterToFileResult(file, null).then((result) => {
            expect(result).toBeFalsy();
        });
    });

    it('testFilterToFileResult_IsHiddenIncludeHidden_True', () => {
        const settings = getSettings();
        settings.includeHidden = true;
        const finder = new Finder(settings);
        const file = '.gitignore';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeTruthy();
        });
    });

    it('testFilterToFileResult_ArchiveNoFindArchives_False', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        const file = 'archive.zip';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeFalsy();
        });
    });

    it('testFilterToFileResult_ArchiveIncludeArchives_True', () => {
        const settings = getSettings();
        settings.includeArchives = true;
        const finder = new Finder(settings);
        const file = 'archive.zip';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeTruthy();
        });
    });

    it('testFilterToFileResult_IsArchiveFindFile_True', () => {
        const settings = getSettings();
        settings.includeArchives = true;
        settings.addInArchiveExtensions('zip');
        const finder = new Finder(settings);
        const file = 'archive.zip';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeTruthy();
        });
    });

    it('testFilterToFileResult_NotIsArchiveFindFile_False', () => {
        const settings = getSettings();
        settings.addOutExtensions('zip');
        const finder = new Finder(settings);
        const file = 'archive.zip';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeFalsy();
        });
    });

    it('testFilterToFileResult_ArchiveFileArchivesOnly_True', () => {
        const settings = getSettings();
        settings.archivesOnly = true;
        expect(settings.archivesOnly).toBeTruthy();
        expect(settings.includeArchives).toBeTruthy();
        const finder = new Finder(settings);
        const file = 'archive.zip';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeTruthy();
        });
    });

    it('testFilterToFileResult_NoExtensionsNoPatterns_True', () => {
        const settings = getSettings();
        const finder = new Finder(settings);
        const file = 'FileUtil.cs';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeTruthy();
        });
    });

    it('testFilterToFileResult_IsFindFile_True', () => {
        const settings = getSettings();
        settings.addInExtensions('cs');
        const finder = new Finder(settings);
        const file = 'FileUtil.cs';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeTruthy();
        });
    });

    it('testFilterToFileResult_NotIsFindFile_False', () => {
        const settings = getSettings();
        settings.addOutExtensions('cs');
        const finder = new Finder(settings);
        const file = 'FileUtil.cs';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeFalsy();
        });
    });

    it('testFilterToFileResult_NonArchiveFileArchivesOnly_False', () => {
        const settings = getSettings();
        settings.archivesOnly = true;
        const finder = new Finder(settings);
        const file = 'FileUtil.cs';
        finder.filterToFileResult(file).then((result) => {
            expect(result).toBeFalsy();
        });
    });

    /*************************************************************
     * test filtering symlink files
     *************************************************************/
    it('test_default_no_symlinks', () => {
        let settings = new FindSettings();
        let binPath = path.join(config.XFIND_PATH, 'bin');
        settings.paths.push(binPath);
        const finder = new Finder(settings);
        finder.find().then(fileResults => {
            expect(fileResults.length < 4).toBeTruthy();
        });
    });

    it('test_follow_symlinks', () => {
        let settings = new FindSettings();
        let binPath = path.join(config.XFIND_PATH, 'bin');
        settings.paths.push(binPath);
        settings.followSymlinks = true;
        const finder = new Finder(settings);
        finder.find().then(fileResults => {
            expect(fileResults.length === 0 || fileResults.length > 2).toBeTruthy();
        });
    });

    it('test_no_follow_symlinks', () => {
        let settings = new FindSettings();
        let binPath = path.join(config.XFIND_PATH, 'bin');
        settings.paths.push(binPath);
        settings.followSymlinks = false;
        const finder = new Finder(settings);
        finder.find().then(fileResults => {
            expect(fileResults.length < 4).toBeTruthy();
        });
    });
});
